src/wayland/seat/pointer.rs

```diff
@@ -226,6 +235,17 @@ impl PointerHandle {
     }
 }
 
+/// Data about the event that started the grab.
+#[derive(Clone)]
+pub struct GrabStartData {
+    /// The focused surface and its location, if any, at the start of the grab.
+    pub focus: Option<(WlSurface, (f64, f64))>,
+    /// The button that initiated the grab.
+    pub button: u32,
+    /// The location of the click that initiated the grab.
+    pub location: (f64, f64),
+}
+
```

Comment:
One last thing, I think it'd be best to make explicit in the doc that the locations here are expressed in *compositor space* (rather than relative to some surface).